e2e/node/wac.test.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  jest,
+  describe,
+  it,
+  expect,
+  beforeEach,
+  afterEach,
+} from "@jest/globals";
+
+import { Session } from "@inrupt/solid-client-authn-node";
+import { config } from "dotenv-flow";
+import {
+  saveSolidDatasetAt,
+  getSolidDatasetWithAcl,
+  hasResourceAcl,
+  getPublicAccess,
+  getAgentAccess,
+  getFallbackAcl,
+  getResourceAcl,
+  getAgentResourceAccess,
+  setAgentResourceAccess,
+  saveAclFor,
+  createSolidDataset,
+  deleteSolidDataset,
+} from "../../src/index";
+// Functions from this module have to be imported from the module directly,
+// because their names overlap with access system-specific versions,
+// and therefore aren't exported from the package root:
+import {
+  getAgentAccess as getAgentAccessUniversal,
+  setPublicAccess as setPublicAccessUniversal,
+} from "../../src/access/universal";
+
+import { getTestingEnvironment, TestingEnvironment } from "./util/getTestingEnvironment";
+import { supportsWac } from "./util/supportsWac";
+import { getSession } from "./util/getSession";
+
+let env: TestingEnvironment;
+
+beforeAll(() => {
+  config({
+    path: __dirname,
+    // Disable warning messages in CI
+    silent: process.env.CI === "true",
+  });
+  env = getTestingEnvironment();
+});
+
+describe(`Authenticated end-to-end WAC tests against environment ${env.environment}`, () => {
+  // Skip test for servers that don't support WAC
+  if (!supportsWac(env.pod)) {
+    return;
+  }
+
+  let options: any;
+  let session: Session;
+  let sessionResource: string;
+
+  beforeEach(async () => {
+    session = await getSession(env);
```

Comment:
Could  `getSession` be renamed `getAuthenticatedSession`for clarity ?

Reply:
Fixed in https://github.com/inrupt/solid-client-js/commit/7b57210bb326483ec569ba19673520a658d0cd19